odxtools/diagservice.py

```diff
@@ -1,50 +1,66 @@
 # SPDX-License-Identifier: MIT
 from dataclasses import dataclass
-from typing import TYPE_CHECKING, Any, Dict, Iterable, List, Optional, Union
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union, cast
 from xml.etree import ElementTree
 
-from .admindata import AdminData
-from .audience import Audience
-from .createsdgs import create_sdgs_from_et
-from .element import IdentifiableElement
-from .exceptions import DecodeError, odxassert, odxrequire
-from .functionalclass import FunctionalClass
+from .communicationparameterref import CommunicationParameterRef
+from .diagcomm import DiagComm
+from .exceptions import DecodeError, odxassert, odxraise, odxrequire
 from .message import Message
 from .nameditemlist import NamedItemList
 from .odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkId, OdxLinkRef
-from .odxtypes import ParameterValue
+from .odxtypes import ParameterValue, odxstr_to_bool
 from .parameters.parameter import Parameter
 from .request import Request
 from .response import Response
-from .specialdatagroup import SpecialDataGroup
-from .state import State
-from .statetransition import StateTransition
 from .utils import dataclass_fields_asdict
 
 if TYPE_CHECKING:
     from .diaglayer import DiagLayer
 
 
+class Addressing(Enum):
+    FUNCTIONAL = "FUNCTIONAL"
+    PHYSICAL = "PHYSICAL"
+    FUNCTIONAL_OR_PHYSICAL = "FUNCTIONAL-OR-PHYSICAL"
+
+
+class TransMode(Enum):
+    SEND_ONLY = "SEND-ONLY"
+    RECEIVE_ONLY = "RECEIVE-ONLY"
+    SEND_AND_RECEIVE = "SEND-AND-RECEIVE"
+    SEND_OR_RECEIVE = "SEND-OR-RECEIVE"
+
+
 @dataclass
-class DiagService(IdentifiableElement):
+class DiagService(DiagComm):
     """Representation of a diagnostic service description.
     """
 
+    comparam_refs: NamedItemList[CommunicationParameterRef]
+
     request_ref: OdxLinkRef
     pos_response_refs: List[OdxLinkRef]
     neg_response_refs: List[OdxLinkRef]
-    admin_data: Optional[AdminData]
-    semantic: Optional[str]
-    audience: Optional[Audience]
-    functional_class_refs: Iterable[OdxLinkRef]
-    pre_condition_state_refs: Iterable[OdxLinkRef]
-    state_transition_refs: Iterable[OdxLinkRef]
-    sdgs: List[SpecialDataGroup]
+
+    #pos_response_suppressable: Optional[PosResponseSuppressable]
```

Comment:
Commented out code?

Reply:
this is meant as a nagging TODO item for anyone who stumbles over this. I prefixed it with "TODO": f5397ea7297731d89757e62c8c75705ed1fc25f8, thanks for the catch!